### ISSUE TITLE
tf.image.resize with bilinear breaks mixed precision

### ISSUE THREAD

**Issue by edwardyehuang** [tensorflow/tensorflow#41934]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow):YES
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04):Ubuntu 18.04
- TensorFlow installed from (source or binary):conda
- TensorFlow version (use command below):2.1/2.2/2.3
- Python version:3.7
- CUDA/cuDNN version:10.1
- GPU model and memory:v100 32GB

Use tf.image.resize with BILINEAR upsampling breaks mixed precision and makes the speed much slower than fp32. Change to NEAREST_NEIGHBOR works

**Comment by Saduf2019** [comment]
@edwardyehuang 
Please update the issue template with exact tensorflow version and sample code for us to replicate the issue faced or if possible share colab gist with error to analyse.

**Comment by edwardyehuang** [comment]
@Saduf2019 

https://colab.research.google.com/drive/1G78SCjm9UVvZGHsqM6ki4LfdT-1dHDJJ?usp=sharing

Make sure you are using Tesla T4 in colab to test mixed precision.

Adjust "ENABLE_MIXED_PRECISION" to verify the effect.

FP32 2s/img vs FP16 4s/img

**Comment by zaccharieramzi** [comment]
@reedwm I see that you fixed this for tf 2.4 which is great.
I was seeing the same effect in my networks and I hadn't pinned it down to upsampling.

Another problem I am noticing is that after a certain (large) number of iterations, nans start appearing in my trainig (I haven't figured out where first yet). Could this be related and should this fix the nan issue?

**Comment by reedwm** [comment]
@zaccharieramzi this fix does not affect NaNs or numerical stability, so unfortunately it's unlikely the NaN issue will be fixed. Unfortunately, NaN issues that only appear after a certain number of iterations are difficult to debug. Try making the loss of the model run in float32 if it doesn't already. Additionally, if the model ends in softmax, run the softmax in float32 as well. You can also try saving a checkpoint on the step before you get a NaN. Then you will be able to load the checkpoint and reproduce the NaN by just running one step.
